packages/amplify-graphql-default-value-transformer/src/graphql-default-value-transformer.ts

```diff
@@ -0,0 +1,162 @@
+
+  private createVTLSnippet = (fieldName: string, defaultValue: string, isString: boolean): string => {
+    return printBlock(`Setting "${fieldName}" to default value of "${defaultValue}"`)(
+      qref(methodCall(ref('ctx.stash.defaultValues.put'), str(fieldName), isString ? str(defaultValue) : raw(defaultValue))),
+    );
+  };
+
+  private augmentResolver = (ctx: TransformerContextProvider, resolverLogicalId: string, snippets: string[]): void => {
+    const resolver = this.getResolverObject(ctx, resolverLogicalId);
+    if (resolver) {
+      this.addSnippetToResolverSlot(resolver, snippets);
+    }
+  };
+
+  private getResolverObject = (ctx: TransformerContextProvider, resolverLogicalId: string): TransformerResolverProvider | null => {
+    const objectName = ctx.output.getMutationTypeName();
+
+    if (!objectName) {
+      return null;
+    }
+
+    return ctx.resolvers.getResolver(objectName, resolverLogicalId) ?? null;
+  };
+
+  private addSnippetToResolverSlot = (resolver: TransformerResolverProvider, snippets: string[]): void => {
+    const res = resolver as any;
+    res.addToSlot(
+      'postAuth',
+      MappingTemplate.s3MappingTemplateFromString(
+        snippets.join('\n') + '\n{}',
+        `${res.typeName}.${res.fieldName}.{slotName}.{slotIndex}.req.vtl`,
+      ),
+    );
+  };
+}
+
+function validate(ctx: TransformerSchemaVisitStepContextProvider, config: DefaultValueDirectiveConfiguration): void {
+  assertModelDirective(config);
+  assertCompatibleFieldType(ctx, config.field.type);
+  assertValidDirectiveArguments(config.directive);
+  assertFieldTypeAndDefaultValueTypeMatch(ctx, config);
+}
+
+function assertCompatibleFieldType(ctx: TransformerSchemaVisitStepContextProvider, type: TypeNode): void {
+  const enums = ctx.output.getTypeDefinitionsOfKind(Kind.ENUM_TYPE_DEFINITION) as EnumTypeDefinitionNode[];
+  if (isListType(type) || !isScalarOrEnum(type, enums)) {
+    throw new InvalidDirectiveError('The @default directive may only be added to scalar or enum field types.');
+  }
+}
+
+function assertValidDirectiveArguments(directive: DirectiveNode): void {
+  if (directive.arguments!.length == 0) throw new InvalidDirectiveError('Directive for @default must declare a value property');
+  if (directive.arguments!.length > 1) throw new InvalidDirectiveError('Directive for @default only takes a value property');
+}
+
+function assertModelDirective(config: DefaultValueDirectiveConfiguration): void {
+  const modelDirective = config.object.directives!.find(dir => dir.name.value === 'model');
+  if (!modelDirective) {
+    throw new InvalidDirectiveError('The @default directive may only be added to object definitions annotated with @model.');
+  }
+}
+
+function assertFieldTypeAndDefaultValueTypeMatch(
+  ctx: TransformerSchemaVisitStepContextProvider,
+  config: DefaultValueDirectiveConfiguration,
+): void {
+  // if field type is non-nullable, ensure value is not null
+  if (config.value == null) {
```

Comment:
Unless you're using `==` to check for both `null` and `undefined` simultaneously, please stick with `===`. You're using `==` elsewhere in this PR.